Build Failure on refs/heads/main

@gemini-code-assist /jules debug this build error

**Log:**
```
/home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/libs/opencv/sdk/native/3rdparty/libs/arm64-v8a/libcpufeatures.a  /home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/libs/opencv/sdk/native/3rdparty/libs/arm64-v8a/libittnotify.a  -ldl  -lm  -llog  /home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/libs/opencv/sdk/native/3rdparty/libs/arm64-v8a/libtegra_hal.a  /home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/libs/opencv/sdk/native/3rdparty/libs/arm64-v8a/libkleidicv_hal.a  /home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/libs/opencv/sdk/native/3rdparty/libs/arm64-v8a/libkleidicv_thread.a  /home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/libs/opencv/sdk/native/3rdparty/libs/arm64-v8a/libkleidicv.a  -latomic -lm && :
  ld.lld: error: undefined symbol: MobileGS::resetGlContext()
  >>> referenced by GraffitiJNI.cpp:75 (/home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/src/main/cpp/GraffitiJNI.cpp:75)
  >>>               CMakeFiles/graffitixr.dir/GraffitiJNI.cpp.o:(Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeResetGlContext)
  
  ld.lld: error: undefined symbol: MobileGS::updateMappingCamera(float*, float*)
  >>> referenced by GraffitiJNI.cpp:139 (/home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/src/main/cpp/GraffitiJNI.cpp:139)
  >>>               CMakeFiles/graffitixr.dir/GraffitiJNI.cpp.o:(Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeUpdateCamera)
  
  ld.lld: error: undefined symbol: MobileGS::updateLightLevel(float)
  >>> referenced by GraffitiJNI.cpp:156 (/home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/src/main/cpp/GraffitiJNI.cpp:156)
  >>>               CMakeFiles/graffitixr.dir/GraffitiJNI.cpp.o:(Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeUpdateLightLevel)
  
  ld.lld: error: undefined symbol: MobileGS::loadSuperPoint(std::__ndk1::vector<unsigned char, std::__ndk1::allocator<unsigned char>> const&)
  >>> referenced by GraffitiJNI.cpp:367 (/home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/src/main/cpp/GraffitiJNI.cpp:367)
  >>>               CMakeFiles/graffitixr.dir/GraffitiJNI.cpp.o:(Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeLoadSuperPoint)
  
  ld.lld: error: undefined symbol: gLastSplatTrace
  >>> referenced by GraffitiJNI.cpp:515 (/home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/src/main/cpp/GraffitiJNI.cpp:515)
  >>>               CMakeFiles/graffitixr.dir/GraffitiJNI.cpp.o:(Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeGetLastSplatTrace)
  >>> referenced by GraffitiJNI.cpp:515 (/home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/src/main/cpp/GraffitiJNI.cpp:515)
  >>>               CMakeFiles/graffitixr.dir/GraffitiJNI.cpp.o:(Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeGetLastSplatTrace)
  clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.
  
  C++ build system [build] failed while executing:
      /usr/local/lib/android/sdk/cmake/3.22.1/bin/ninja \
        -C \
        /home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge/.cxx/Debug/4d5m5x6h/arm64-v8a \
        graffitixr
    from /home/runner/work/GraffitiXR/GraffitiXR/core/nativebridge

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights from a Build Scan (powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.4.0-rc-1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 39s
275 actionable tasks: 142 executed, 133 from cache

```

--- a/core/nativebridge/src/main/cpp/GraffitiJNI.cpp
+++ b/core/nativebridge/src/main/cpp/GraffitiJNI.cpp
@@ -11,7 +11,7 @@
 #define LOGD(...) __android_log_print(ANDROID_LOG_DEBUG, "GraffitiJNI", __VA_ARGS__)
 
 static std::string gLastDepthTrace;
-extern std::string gLastSplatTrace;
+static std::string gLastSplatTrace;
 #define DEPTH_TRACE(fmt, ...) do {     char _buf[256];     snprintf(_buf, sizeof(_buf), fmt, ##__VA_ARGS__);     LOGD("DEPTH_PIPE: %s", _buf);     gLastDepthTrace += std::string(_buf) + "\n"; } while(0)
 #define LOGE(...) __android_log_print(ANDROID_LOG_ERROR, "GraffitiJNI", __VA_ARGS__)
 

--- a/core/nativebridge/src/main/cpp/MobileGS.cpp
+++ b/core/nativebridge/src/main/cpp/MobileGS.cpp
@@ -166,6 +166,7 @@ void MobileGS::updateMappingCamera(float* viewMat, float* projMat) {
 }
 
 void MobileGS::updateLightLevel(float level) {
+    std::lock_guard<std::mutex> lock(mMutex);
     mLightLevel = level;
 }
 

--- a/core/nativebridge/src/main/cpp/MobileGS.cpp
+++ b/core/nativebridge/src/main/cpp/MobileGS.cpp
@@ -63,6 +63,10 @@ void MobileGS::initGl() {
     mSurfaceMesh.initGl();
 }
 
+void MobileGS::resetGlContext() {
+    initGl();
+}
+
 void MobileGS::draw() {
     std::lock_guard<std::mutex> lock(mMutex);
     interpolateAnchorStep();
@@ -155,6 +159,16 @@ void MobileGS::updateCamera(float* viewMat, float* projMat) {
     mCameraReady = true;
 }
 
+void MobileGS::updateMappingCamera(float* viewMat, float* projMat) {
+    std::lock_guard<std::mutex> lock(mMutex);
+    memcpy(mMappingViewMatrix, viewMat, 16 * sizeof(float));
+    memcpy(mMappingProjMatrix, projMat, 16 * sizeof(float));
+}
+
+void MobileGS::updateLightLevel(float level) {
+    mLightLevel = level;
+}
+
 void MobileGS::updateAnchorTransform(float* transformMat) {
     std::lock_guard<std::mutex> lock(mMutex);
     memcpy(mAnchorMatrix, transformMat, 16 * sizeof(float));
@@ -190,3 +204,4 @@ void MobileGS::restoreWallFingerprint(const cv::Mat& d, const std::vector<cv::Po
 void MobileGS::scheduleRelocCheck(const cv::Mat& f) { mRelocColorFrame = f.clone(); mRelocRequested = true; mRelocCv.notify_one(); }
 void MobileGS::setArtworkFingerprint(const cv::Mat& c, const uint8_t* d, int w, int h, int s, const float* i, const float* v) {}
 MobileGS::FingerprintData MobileGS::generateFingerprint(const cv::Mat& i, const cv::Mat& m, const uint8_t* d, int w, int h, int s, const float* intr, const float* v) { return {}; }
+bool MobileGS::loadSuperPoint(const std::vector<uchar>& onnxBytes) { return mSuperPoint.load(onnxBytes); }

--- a/core/nativebridge/src/main/cpp/include/MobileGS.h
+++ b/core/nativebridge/src/main/cpp/include/MobileGS.h
@@ -174,6 +174,7 @@ private:
     bool mPersistentMeshInitialized = false;
 
     uint64_t mFrameCounter = 0;
+    float mLightLevel = 1.0f;
 
     float mViewMatrix[16];
     float mProjMatrix[16];

